### PR TITLE
Allow managers to send announcements

### PR DESCRIPTION
**Pull request by lucasleite01**
Managers should also be allowed to send announcements, since they can have many users bellow them to manage.

Plus, this would enhance the announcements feature to cover scenarios of Traccar servers where managers want to send information to their managed users.

### CODE REVIEW

**Comment by tananaev** [comment]
Do you have a corresponding web app change?

**Comment by lucasleite01** [comment]
Yes, check [here](https://github.com/traccar/traccar-web/pull/1268) the corresponding web app change.

File: src/main/java/org/traccar/api/resource/NotificationResource.java [hunk 1]
```diff
@@ -113,15 +114,27 @@ public Response testMessage(@PathParam("notificator") String notificator)
...
+                if (permissionsService.notAdmin(getUserId())) {
+                    var conditions = new LinkedList<Condition>();
+                    conditions.add(new Condition.Equals("id", userId));
+                    conditions.add(new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups());
+                    users.add(storage.getObject(
+                            User.class, new Request(new Columns.All(), Condition.merge(conditions))));
+                } else {
+                    users.add(storage.getObject(
+                            User.class, new Request(new Columns.All(), new Condition.Equals("id", userId))));
+                }
```
Comment (tananaev) [inline src/main/java/org/traccar/api/resource/NotificationResource.java:137]: It probably makes sense to do something like this:

```suggestion
                    var conditions = new LinkedList<Condition>();
                    conditions.add(new Condition.Equals("id", userId));
                if (permissionsService.notAdmin(getUserId())) {
                    conditions.add(new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups());
                }
                    users.add(storage.getObject(
                            User.class, new Request(new Columns.All(), Condition.merge(conditions))));
```

File: src/main/java/org/traccar/api/resource/NotificationResource.java [hunk 2]
```diff
@@ -113,15 +114,27 @@ public Response testMessage(@PathParam("notificator") String notificator)
...
             NotificationMessage message) throws MessageException, StorageException {
-        permissionsService.checkAdmin(getUserId());
+        permissionsService.checkManager(getUserId());
         List<User> users;
         if (userIds.isEmpty()) {
-            users = storage.getObjects(User.class, new Request(new Columns.All()));
+            users = permissionsService.notAdmin(getUserId()) ?
+                    storage.getObjects(User.class, new Request(new Columns.All(),
+                            new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups())) :
+                    storage.getObjects(User.class, new Request(new Columns.All()));
```
Comment (tananaev) [inline src/main/java/org/traccar/api/resource/NotificationResource.java:123]: Let's do if/else here. Tertiary expression we mostly use for short things that make sense to inline.

File: src/main/java/org/traccar/api/resource/NotificationResource.java [hunk 3]
```diff
@@ -113,15 +114,28 @@ public Response testMessage(@PathParam("notificator") String notificator)
...
+                var conditions = new LinkedList<Condition>();
+                conditions.add(new Condition.Equals("id", userId));
+                if (permissionsService.notAdmin(getUserId())) {
+                    conditions.add(new Condition.Permission(User.class, getUserId(), ManagedUser.class).excludeGroups());
+                    users.add(storage.getObject(
+                            User.class, new Request(new Columns.All(), Condition.merge(conditions))));
+                } else {
+                    users.add(storage.getObject(
+                            User.class, new Request(new Columns.All(), Condition.merge(conditions))));
+                }
```
Comment (tananaev) [inline src/main/java/org/traccar/api/resource/NotificationResource.java:138]: This doesn't look like what I suggested. Why do we need to duplicate `users.add(storage.getObject(...`?

**Comment by tananaev** [comment]
Make sure you compile and test it. Currently build is failing.

**Comment by lucasleite01** [comment]
The error was a style check:
image

The last commit fixes it.

**Pull request merged**